crates/primitives/src/signature/parity.rs

```diff
@@ -76,6 +76,25 @@ impl Parity {
         }
     }
 
+    /// Applies a check to the [`Parity::Eip155`] v value if this instance is a [`Parity::Eip155`].
+    ///
+    /// Returns [`Ok`] if the instance is not [`Parity::Eip155`]
+    pub fn ensure_eip155<F, E>(&self, f: F) -> Result<(), E>
```

Comment:
This is too specific, doesn't make sense to have as a method, `is_` and `as_` are enough

Reply:
hmm this would make things like:


```
            // Check y_parity
            if let Parity::Eip155(parity) = auth.signature().v() {
                if parity > u8::MAX as u64 {
                    return Err(InvalidAuthorization::InvalidYParity);
                }
            }
            Ok(())
```

slightly easier imo

Reply:
the same is achievable with as_eip().is_some_and(), again this is too specific

Reply:
still think this is nice to have and doesn't hurt really, but closing because can be solved with some extra calls as well